### PR TITLE
add fpr & tpr aggregation and ks & auc computation

### PR DESCRIPTION
**Pull request by xuefeng-xu**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](https://github.com/primihub/community#development-workflow) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] Please provide a readable PR title.

### Description
Please describe what have you done in this pull request.
- aggregate fpr & tpr from two clients to get the global metrics
- compute ks & auc from fpr & tpr arrays

### CODE REVIEW

**Review by fuxingbit — COMMENTED** [review]
!

File: python/primihub/FL/model/metrics/metrics.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                         weights=num_negtive_examples_weights)
+        tpr = np.average([tpr1_prev, tpr2_prev],
+                         weights=num_positive_examples_weights)
+        fpr_merge.append(fpr)
+        tpr_merge.append(tpr)
+
+    return fpr_merge, tpr_merge, thresholds_merge
+
+
+def ks_from_fpr_tpr(fpr, tpr):
```
Comment (fuxingbit) [inline python/primihub/FL/model/metrics/metrics.py:49]: dfdsf

File: python/primihub/FL/model/metrics/metrics.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import numpy as np
+
+
+def fpr_tpr_merge2(fpr1, tpr1, thresholds1,
```
Comment (xujiangyu) [inline python/primihub/FL/model/metrics/metrics.py:4]: Please add the referrence docs

File: python/primihub/FL/model/metrics/metrics.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                   num_positive_examples_weights,
+                   num_negtive_examples_weights):
+    fpr_merge = []
+    tpr_merge = []
+    thresholds_merge = []
+
+    idx1, idx2 = 0, 0
+    fpr1_prev, tpr1_prev, fpr2_prev, tpr2_prev = 0, 0, 0, 0
+
+    while idx1 < len(thresholds1) or idx2 < len(thresholds2):
```
Comment (xujiangyu) [inline python/primihub/FL/model/metrics/metrics.py:15]: Add comments notes

File: python/primihub/FL/model/logistic_regression/homo_lr_dev.py [hunk 4]
```diff
@@ -299,27 +301,50 @@ def broadcast_global_model_param(self):
...
+        acc = np.average(accs, weights=num_examples_weights)
+
+        # fpr, tpr
+        host_fpr = self.host_channel.recv("host_fpr")
+        guest_fpr = self.guest_channel.recv("guest_fpr")
+        host_tpr = self.host_channel.recv("host_tpr")
+        guest_tpr = self.guest_channel.recv("guest_tpr")
+        host_thresholds = self.host_channel.recv("host_thresholds")
+        guest_thresholds = self.guest_channel.recv("guest_thresholds")
+        fpr, tpr, thresholds = fpr_tpr_merge2(host_fpr, host_tpr, host_thresholds,
```
Comment (xujiangyu) [inline python/primihub/FL/model/logistic_regression/homo_lr_dev.py:328]: whether it has the problem of information leakage?

File: python/primihub/FL/model/metrics/metrics.py (hunk 2 above)
Comment (xuefeng-xu) [reply python/primihub/FL/model/metrics/metrics.py:5]: merge fpr and tpr from two clients

File: python/primihub/FL/model/metrics/metrics.py (hunk 3 above)
Comment (xuefeng-xu) [reply python/primihub/FL/model/metrics/metrics.py:17]: merge two descending thresholds arrays

File: python/primihub/FL/model/logistic_regression/homo_lr_dev.py (hunk 4 above)
Comment (xuefeng-xu) [reply python/primihub/FL/model/logistic_regression/homo_lr_dev.py:328]: Yes, the fpr, tpr, and thresholds arrarys do have privacy issues, but here we assume the arbiter is a trusted third party.

**Review by xujiangyu — APPROVED** [review]

**Pull request merged**